src/Header.test.jsx

```diff
@@ -101,4 +113,79 @@ describe('<Header />', () => {
 
     expect(wrapper.toJSON()).toMatchSnapshot();
   });
+
+  it('renders exam dashboard link when ENABLE_EXAM_DASHBOARD is true (desktop)', () => {
+    const contextValue = {
+      authenticatedUser: {
+        userId: 'abc123',
+        username: 'edX',
+        roles: [],
+        administrator: false,
+      },
+      config: {
+        LMS_BASE_URL: process.env.LMS_BASE_URL,
+        SITE_NAME: process.env.SITE_NAME,
+        LOGIN_URL: process.env.LOGIN_URL,
+        LOGOUT_URL: process.env.LOGOUT_URL,
+        LOGO_URL: process.env.LOGO_URL,
+        ENABLE_EXAM_DASHBOARD: true,
+        EXAM_DASHBOARD_MFE_BASE_URL: 'http://localhost:1994',
+      },
+    };
+    const component = <HeaderComponent width={{ width: 1280 }} contextValue={contextValue} />;
+
+    const wrapper = TestRenderer.create(component);
+
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders exam dashboard link when ENABLE_EXAM_DASHBOARD is true (mobile)', () => {
+    const contextValue = {
+      authenticatedUser: {
+        userId: 'abc123',
+        username: 'edX',
+        roles: [],
+        administrator: false,
+      },
+      config: {
+        LMS_BASE_URL: process.env.LMS_BASE_URL,
+        SITE_NAME: process.env.SITE_NAME,
+        LOGIN_URL: process.env.LOGIN_URL,
+        LOGOUT_URL: process.env.LOGOUT_URL,
+        LOGO_URL: process.env.LOGO_URL,
+        ENABLE_EXAM_DASHBOARD: true,
+        EXAM_DASHBOARD_MFE_BASE_URL: 'http://localhost:1994',
+      },
+    };
+    const component = <HeaderComponent width={{ width: 500 }} contextValue={contextValue} />;
+
+    const wrapper = TestRenderer.create(component);
+
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('does not render exam dashboard link when ENABLE_EXAM_DASHBOARD is false', () => {
+    const contextValue = {
+      authenticatedUser: {
+        userId: 'abc123',
+        username: 'edX',
+        roles: [],
+        administrator: false,
+      },
+      config: {
+        LMS_BASE_URL: process.env.LMS_BASE_URL,
+        SITE_NAME: process.env.SITE_NAME,
+        LOGIN_URL: process.env.LOGIN_URL,
+        LOGOUT_URL: process.env.LOGOUT_URL,
+        LOGO_URL: process.env.LOGO_URL,
+        ENABLE_EXAM_DASHBOARD: false,
+      },
+    };
+    const component = <HeaderComponent width={{ width: 1280 }} contextValue={contextValue} />;
+
+    const wrapper = TestRenderer.create(component);
+
+    // Verify that the exam dashboard link is not present in the snapshot
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
```

Comment:
Not sure about these tests, I do not feel we are testing properly, how exactly we determine if an element exist or don't? All are just snapshots, maybe something more accurate could be:


```jsx

it('does not render exam dashboard link when ENABLE_EXAM_DASHBOARD is false', () => {
  const contextValue = {
    authenticatedUser: {
      userId: 'abc123',
      username: 'edX',
      roles: [],
      administrator: false,
    },
    config: {
      ENABLE_EXAM_DASHBOARD: false,
    },
  };

  const wrapper = TestRenderer.create(
    <HeaderComponent width={{ width: 1280 }} contextValue={contextValue} />
  );

  const tree = wrapper.toJSON();
  const link = JSON.stringify(tree);

  expect(link).not.toContain('exam-dashboard');
});

```


But let me think what do you think.